oauthlib/oauth2/rfc6749/endpoints/metadata.py

```diff
@@ -89,21 +89,25 @@ def validate_metadata(self, array, key, is_required=False, is_list=False, is_url
                     raise ValueError("array {}: {} must contains only string (not {})".format(key, array[key], elem))
 
     def validate_metadata_token(self, claims, endpoint):
-        claims.setdefault("grant_types_supported", list(endpoint._grant_types.keys()))
+        self._grant_types += list(endpoint._grant_types.keys())
```

Comment:
`self._grant_types.extend(endpoint._grant_types.keys())` is clearer in my opinion and doesn't require initializing a new list out of the keys view.